compiler/rustc_ty_utils/src/implied_bounds.rs

```diff
@@ -9,12 +9,12 @@ pub fn provide(providers: &mut ty::query::Providers) {
 fn assumed_wf_types(tcx: TyCtxt<'_>, def_id: DefId) -> &ty::List<Ty<'_>> {
     match tcx.def_kind(def_id) {
         DefKind::Fn => {
-            let sig = tcx.fn_sig(def_id);
+            let sig = tcx.fn_sig(def_id).skip_binder();
             let liberated_sig = tcx.liberate_late_bound_regions(def_id, sig);
             liberated_sig.inputs_and_output
         }
         DefKind::AssocFn => {
-            let sig = tcx.fn_sig(def_id);
+            let sig = tcx.fn_sig(def_id).skip_binder();
```

Comment:
I think what we do here is closer to `subst_identity` as inside of that function we actually "are" inside of the binder and actually don't even exit the binder at the end of this function as we still have the liberated late bound regions which would be wrong when used outside of the function itself

Reply:
Ok, sounds good. Guessing this is also true for the first match arm (`DefKind::Fn`)?